### PR TITLE
reduce timeouts on network calls during logout

### PR DESCRIPTION
**Pull request by heronhaye**

### CODE REVIEW

File: go/engine/login_state_test.go [hunk 1]
```diff
@@ -361,14 +361,14 @@ type timeoutAPI struct {
 }
 
 func (r *timeoutAPI) GetDecode(mctx libkb.MetaContext, arg libkb.APIArg, w libkb.APIResponseWrapper) error {
-	return libkb.APINetError{}
+	return libkb.APINetError{Err: errors.New("timeout")}
```
Comment (jakob223) [inline go/engine/login_state_test.go]: make a var for this err plz (`errFakeNetworkTimeout` or somthing)

File: go/libkb/logout.go [hunk 2]
```diff
@@ -176,6 +177,8 @@ func CanLogout(mctx MetaContext) (res keybase1.CanLogoutRes) {
 		return res
 	}
 
+	mctx, cancel := mctx.WithTimeout(5 * time.Second)
```
Comment (heronhaye) [inline go/libkb/logout.go]: cc @patrickxb, I'm putting at imeout on this, otherwise it stalls for 20-30+ seconds polling merkle. Sound ok?

Comment (patrickxb) [reply go/libkb/logout.go]: You mean if you're offline?

Comment (heronhaye) [reply go/libkb/logout.go]: Yes

Comment (heronhaye) [reply go/libkb/logout.go]: ok merging this, lmk if you want me to back it out.

**Comment by heronhaye** [comment]
merging over ci flake

**Pull request merged**